### ISSUE TITLE
Border misaligned when containing some unicode characters

### ISSUE THREAD

**Issue by zephraph** [vadimdemedes/ink#356]
I'm playing around with making an interface for playing card games and an issue that I'm running into is that when the ♠︎ character (or any card suit) is rendered, the border is collapse. See below. 

<img width="82" alt="image" src="https://user-images.githubusercontent.com/3087225/89444439-b17def80-d71f-11ea-8757-3d7bcc5ce655.png">

I've tired it in the regular terminal app, hyper, and vscode's terminal all to the same result. 

This is v3.0.3 on OSX.

**Comment by zephraph** [comment]
I wanted to get a repl.it example of it, but it looks like borders don't render there at all. 

Here's a minimal code repl:

```jsx
const  { render, Box, Text } = require("ink");

function Card(props) {
  return (
    <Box borderStyle="round" width={10} height={7}>
      <Text>A♣︎</Text>
    </Box>
  );
}

render(<Card/>);
```

**Comment by zephraph** [comment]
Okay, so... this isn't what I thought. I started digging deeper and checking through all the length libraries (which seemed to be doing the right thing). 

Turns out when I grabbed the character from OSX's Character Viewer it had some invisible character on the end which could be safely deleted.

Maybe there's something we could do to prevent invisible characters from affecting the layout, but this issue as written isn't really an issue.
